packages/SwingSet/test/test-netstring.js

```diff
@@ -0,0 +1,60 @@
+import '@agoric/install-ses'; // adds 'harden' to global
+import test from 'ava';
+import { encode, decode } from '../src/netstring';
+
+test('encode', t => {
+  function eq(input, expected) {
+    const encoded = encode(Buffer.from(input));
+    const expBuf = Buffer.from(expected);
+    if (encoded.compare(expBuf) !== 0) {
+      console.log(`got : ${encoded}`);
+      console.log(`want: ${expBuf}`);
+    }
+    t.deepEqual(encoded, expBuf);
+  }
+
+  eq('', '0:,');
+  eq('a', '1:a,');
+  eq('abc', '3:abc,');
+  const umlaut = 'ümlaut';
+  t.is(umlaut.length, 6);
+  const umlautBuffer = Buffer.from(umlaut, 'utf-8');
```

Comment:
I’ve been leaning harder on web API’s to make these kinds of libraries easier to port. To that end, `Uint8Array`, `TextDecoder`, and `TextEncoder` are handy, albeit less convenient than `Buffer.from`.

Reply:
It would also increase portability to frame this API in terms of async iterators and relieve the dependence on Node.js-specific streams. Node.js streams do implement async iterators so bridging should not be hard.